### ISSUE TITLE
20.2 → 21.1 roachperf benchmark regressions

### ISSUE THREAD

**Issue by erikgrinaker** [cockroachdb/cockroach#62322]
Tracking issue for 20.2 → 21.1 [roachperf](https://roachperf.crdb.dev) benchmark regressions.

- [x] #62078: tpccbench regression between Feb 12th and 13th
  - **Cause:** separated intents #59829
  - **Partial fix:** cockroachdb/pebble#1091 #62264
  - **Backport:** #62676 
- [x] #62148: investigate kv95 regression around Feb 18
  - **Cause:** always-on tracing #59992
  - **Fix:** #61777 #62227
  - **Backport:** #61965
- [x] #62359: investigate kv95 performance after Feb 18
  - [x] #62447: kvfollowerreadsccl: excessive overhead in canSendToFollower
    - **Cause:** follower reads #59571
    - **Fix:** #62465 #62498
    - **Backport:** #62606
  - [x] catalog `protoutil.Clone()`
    - **Cause:**  #61429
    - **Fix:** #62388
    - **Backport:** #62616
- [ ] ~~#62156: gradual kv95 performance regression since December 1st~~
- [x] #62198: YCSB/E performance regression on December 4th
  - **Cause:** vectorized execution #55713
  - **Fix:** #62282 #62534
  - **Backport:** #62365 #62603
- [x] #62523: YCSB/A performance regression on December 23rd
  - **Cause:** Pebble read compation #58247 cockroachdb/pebble#1032
  - **Fix:** cockroachdb/pebble#1098 #62673
  - **Backport:** cockroachdb/pebble#1101

See also [investigation spreadsheet](https://docs.google.com/spreadsheets/d/1by4iKjPvNFjiMEoUpKEiusd3-Cf2uWFfVr2T6B5eBx4/edit#gid=0).

### Branch Benchmarks

`kv95/enc=false/nodes=1/cpu=32` on GCE, median cumulative ops/s over 5 runs.

| Date | `release-20.2` | `release-21.1` | `master` |
|--|--|--|--|
| 2021-03-25 | 76,018 ops/s | 53,462 ops/s | 70,149 ops/s |
| 2021-03-29 | 76,713 ops/s | 70,672 ops/s | 70,763 ops/s |

### Follow-up tasks

- [x] cockroachdb/roachperf#49: Add ability to manually annotate graphs
- [ ] cockroachdb/roachperf#48: Move roachperf to grafana to improve visualization
- [ ] cockroachdb/roachperf#33: capture commit SHA in test metadata
- [ ] #62309: roachprod: commands for pprof profiling
- [ ] #62303: roachtest: override workload flags
- [ ] #62302: roachtest: use smaller workload machines
- [x] Write wiki guide for pre-release QA audit of benchmarks and performance regressions

**Comment by erikgrinaker** [comment]
We've picked all of the low-hanging fruit at this point. We haven't been able to pinpoint the causes of the gradual decline in performance in the Dec-Feb time frame (#62156), the only remaining known regressions are due to separated intents lock table reads (#62078) and tracing. We did a few runs with these disabled:

* `master`: 70,763 ops/s
* w/o lock table reads: 74,707 ops/s
* w/o lock table reads + tracing: 75,422 ops/s

This roughly gets us back to 20.2. I suspect vectorized execution may make up the remainder -- or just an accumulation of other minor regressions.
